perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -1103,6 +1123,196 @@ public static void sendValidationEmail(User user, String url, String email, int
 	 * @throws InternalErrorException
 	 */
 	public static void sendPasswordResetEmail(User user, String email, String namespace, String url, int id, String messageTemplate, String subject) {
+		String instanceName = BeansUtils.getCoreConfig().getInstanceName();
+
+		String validationLink = prepareValidationLinkForPasswordResetAndAccountActivation(url, "/non/pwd-reset/", id, user, namespace, false);
+		String validityTo = prepareValidityTo();
+
+		String defaultSubject = "[" + instanceName + "] Password reset in namespace: " + namespace;
+		String defaultBody = "Dear " + user.getDisplayName() + ",\n\nWe've received request to reset your password in namespace \"" + namespace + "\"." +
+			"\n\nPlease visit the link below, where you can set new password:\n\n" + validationLink + "\n\n" +
+			"Link is valid till " + validityTo + "\n\n" +
+			"Message is automatically generated." +
+			"\n----------------------------------------------------------------" +
+			"\nPerun - Identity & Access Management System";
+
+		Map<String, String> subjectParametersToReplace = new HashMap<>();
+		subjectParametersToReplace.put("{instanceName}", instanceName);
+		subjectParametersToReplace.put("{namespace}", namespace);
+		subject = prepareSubjectOfEmail(subject, defaultSubject, subjectParametersToReplace);
+
+		Map<String, String> bodyParametersToReplace = new HashMap<>();
+		bodyParametersToReplace.put("{displayName}", user.getDisplayName());
+		bodyParametersToReplace.put("{namespace}", namespace);
+		bodyParametersToReplace.put("{validity}", validityTo);
+		// allow enforcing per-language links
+		if (messageTemplate.contains("{link-")) {
```

Comment:
This fails if ``messageTemplate`` is null, which is, when there is no pre-defined template stored in the entityless attribute. Same for the account activation method.

Reply:
Fixed.